src/ngraph/op/experimental/layers/proposal.cpp

```diff
@@ -35,25 +35,67 @@ op::Proposal::Proposal(const std::shared_ptr<Node>& class_probs,
 
 void op::Proposal::validate_and_infer_types()
 {
-    // shape node should have integer data type. For now we only allow i64
-    auto image_shape_et = get_input_element_type(2);
+    const auto& class_prob_et = get_input_element_type(0);
+    const auto& class_logits_et = get_input_element_type(1);
+    const auto& image_shape_et = get_input_element_type(2);
+
+    NODE_VALIDATION_CHECK(this,
```

Comment:
As I understand we do not usually check type of input tensor, because it can be fp32 or f16 or smth else.

Reply:
I agree. Remove the these type checks.

Reply:
Done